### PR TITLE
Fix bug: signal workflow will trigger cron workflow execution

### PR DESCRIPTION
**Pull request by yux0**
the signal workflow execution should not trigger cron workflow execution

### CODE REVIEW

File: host/signalworkflow_test.go [hunk 1]
```diff
@@ -586,6 +586,66 @@ CheckHistoryLoopForSignalSent:
...
+		WorkflowExecution: &workflow.WorkflowExecution{
+			WorkflowId: common.StringPtr(id),
+			RunId:      common.StringPtr(*we.RunId),
+		},
+		SignalName: common.StringPtr(signalName),
+		Input:      signalInput,
+		Identity:   common.StringPtr(identity),
+	})
+	s.Nil(err)
+	resp, err := s.engine.GetWorkflowExecutionHistory(createContext(), &workflow.GetWorkflowExecutionHistoryRequest{
```
Comment (yycptt) [inline host/signalworkflow_test.go]: I am not sure but will this create a race condition? If you get the history before the decision task for the signal is added to history, then the test can still pass.

Comment (yux0) [reply host/signalworkflow_test.go]: Updated the integration test to check the signal event created but no decision task creation

File: host/signalworkflow_test.go [hunk 2]
```diff
@@ -586,6 +586,71 @@ CheckHistoryLoopForSignalSent:
...
+		Domain: common.StringPtr(s.domainName),
+		Execution: &workflow.WorkflowExecution{
+			WorkflowId: common.StringPtr(id),
+			RunId:      common.StringPtr(*we.RunId),
+		},
+		MaximumPageSize: common.Int32Ptr(1000),
+		WaitForNewEvent: common.BoolPtr(false),
+	})
+	isSignalEventZCreated := false
```
Comment (yycptt) [inline host/signalworkflow_test.go]: isSignalEventCreated

Comment (yux0) [reply host/signalworkflow_test.go]: Deleted after the new test case

File: host/signalworkflow_test.go (hunk 1 above)
Comment (yux0) [reply host/signalworkflow_test.go]: Updated with the new test approach

File: host/signalworkflow_test.go [hunk 3]
```diff
@@ -586,6 +586,121 @@ CheckHistoryLoopForSignalSent:
...
+	s.Nil(err)
+
+	// decider logic
+	activityScheduled := false
+	activityData := int32(1)
+	var decisionTaskDelay time.Duration
+	dtHandler := func(execution *workflow.WorkflowExecution, wt *workflow.WorkflowType,
+		previousStartedEventID, startedEventID int64, history *workflow.History) ([]byte, []*workflow.Decision, error) {
+		decisionTaskDelay = time.Now().Sub(now)
+		if !activityScheduled {
```
Comment (yycptt) [inline host/signalworkflow_test.go]: I don't think you need anything related to activity. Just complete workflow execution directly in the dtHandler should be enough.

File: host/signalworkflow_test.go [hunk 4]
```diff
@@ -586,6 +586,92 @@ CheckHistoryLoopForSignalSent:
...
+		return nil, []*workflow.Decision{{
+			DecisionType: common.DecisionTypePtr(workflow.DecisionTypeCompleteWorkflowExecution),
+			CompleteWorkflowExecutionDecisionAttributes: &workflow.CompleteWorkflowExecutionDecisionAttributes{
+				Result: []byte("Done."),
+			},
+		}}, nil
+	}
+
+	// activity handler
+	atHandler := func(execution *workflow.WorkflowExecution, activityType *workflow.ActivityType,
```
Comment (yycptt) [inline host/signalworkflow_test.go]: This can also be deleted.

**Pull request merged**